src/dialect/mod.rs

```diff
@@ -63,6 +63,152 @@ macro_rules! dialect_of {
     };
 }
 
+/// Constant settings for a dialect.
+///
+/// For configuration to be defined here, rather than on the rate:
+/// * It must be a const setting, not value that's dependent on context - that has to be a function
+/// * have a default that matches `Default::default()`
+#[derive(Debug, Default, Clone)]
+pub struct DialectFlags {
```

Comment:
Would it make sense to call it e.g. `DialectSettings`, `DialectConstants` or similar if we envision this potentially containing more than boolean flags?

Reply:
Happy to change a name.